src/opnsense/scripts/unbound/start.sh

```diff
@@ -26,11 +26,26 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
-set -e
-
 # prepare and startup unbound, so we can easily background it
 
-chroot -u unbound -g unbound / /usr/local/sbin/unbound-anchor -a /var/unbound/root.key
+# if the root.key file is missing or damaged, run unbound-anchor
+/usr/local/sbin/unbound-checkconf /var/unbound/unbound.conf 2> /dev/null 
+
+if [ $? -eq 1 ]; then
+	# unbound-anchor has undefined behaviour if file is corrupted, start clean 
+	rm -f /var/unbound/root.key
+
+	# if we are in forwarding mode, prefer to use the configured system nameservers
+	if [ $(/usr/local/sbin/pluginctl -g unbound.forwarding) ]; then
+		OPT_RESOLVE="-Rf /etc/resolv.conf"
```

Comment:
small note: we don't have to strictly point to /etc/resolv.conf and could populate our ow resolv.conf for bootstrapping (if it makes sense)

Reply:
This is actually a neat suggestion to integrate it with the MVC custom forwarding page. If we could do this earlier in the chain we can also use this to migrate the entire legacy query forwarding settings to MVC. The current hitch is the fact that system nameservers are awkward to fetch in MVC.

Reply:
you mean system: settings: general? if so we are not quite there yet

Reply:
Ok, I'll convert the PR to a draft state and figure out a solution to this and keep you updated.

Reply:
@fichtner changes regarding system: settings: general addressed in https://github.com/opnsense/core/pull/5606